clang/lib/DPCT/RulesAsm/AsmMigration.cpp

```diff
@@ -1093,6 +1093,132 @@ class SYCLGen : public SYCLGenBase {
     return HandleAddSub(Inst);
   }
 
+  bool HandleShfulSyncOp(const InlineAsmInstruction *Inst) {
+
+    if (emitStmt(Inst->getOutputOperand()))
+      return SYCLGenError();
+    OS() << " = ";
+
+    if (DpctGlobalInfo::useMaskedSubGroupFunction()) {
+
+      std::string Op[4];
+      if (tryEmitAllInputOperands(Op, Inst))
+        return SYCLGenError();
+      if (Inst->hasAttr(InstAttr::up)) {
+        // to handle "shfl.up.b32 %0, %1, %2, %3;"
+        OS() << MapNames::getDpctNamespace()
+             << "experimental::" << "shift_sub_group_right(" << Op[3] << ", ";
+      } else if (Inst->hasAttr(InstAttr::down)) {
+        // to handle "shfl.down.b32 %0, %1, %2, %3;"
+        OS() << MapNames::getDpctNamespace()
+             << "experimental::" << "shift_sub_group_left(" << Op[3] << ", ";
+      } else if (Inst->hasAttr(InstAttr::idx)) {
+        // to handle "shfl.sync.idx.b32 %0, %1, %2, %3, %4;"
+        OS() << MapNames::getDpctNamespace()
+             << "experimental::" << "select_from_sub_group(" << Op[3] << ", ";
+      } else if (Inst->hasAttr(InstAttr::bfly)) {
+        // to handle "shfl.bfly.b32 %0, %1, %2, %3;"
+        OS() << MapNames::getDpctNamespace()
+             << "experimental::" << "permute_sub_group_by_xor(" << Op[3]
+             << ", ";
+      }
+
+      OS() << DpctGlobalInfo::getItem(GAS) << ".get_sub_group(), " << Op[0]
+           << ", " << Op[1] << ")";
+    } else {
+      std::string Op[4];
+      if (tryEmitAllInputOperands(Op, Inst))
+        return SYCLGenError();
+
+      std::string WarningMsg =
```

Comment:
I think we can use `const char *` or `StringRef` here to avoid construct a `std::string`.

Reply:
In line1142, std::string  can be used for string concatenation, while StringRef/const char * cannot do that.